### DISCUSSION TITLE
Bitdefender detects trojan in the .jar file

### DISCUSSION THREAD

**Discussion by jkiller295** [jagrosh/MusicBot#958] [Questions] [answered]
I downloaded the JMusicBot-0.3.6.jar file from the release page, and Bitdefender immediately detects a Trojan in the file

![jmusic](https://user-images.githubusercontent.com/4377795/135993758-468ae81c-a657-4f11-9529-e14ac702fdf7.png)

What could trigger this alert?

**Comment by MichailiK** [comment]
[Disable Bitdefender](https://www.bitdefender.com/consumer/support/answer/28557/) or [exclude JMusicBot from Bitdefender](https://www.bitdefender.com/consumer/support/answer/13427/).

We don't (and probably can't) know why Bitdefender is flagging the bot as malicious.

**Comment by hermitoff** [comment]
I simply put this here : https://www.virustotal.com/gui/file/1dcdddb0e0030be8d5652718d7d3980e41e29a7b4c8ddc3f0680771144bf7749

**Reply by jkiller295** [reply]
The strange thing is when I used virustotal with the direct link of the .jar file, it couldn't detect anything; but after downloading it and uploading to virustotal, I got the same result as your link

**Reply by samhwang** [reply]
My findings are:
- Chromium Edge on Windows also blocks it from being downloaded but Chrome does it fine. Also Chromium Edge on Mac doesn't flag the file as well.
- But when I downloaded the file, I couldn't get any errors from scanning the jar file in Windows Defender.

Screenshot when I downloaded the Jar file on Chromium Edge on Windows 10:
![image](https://user-images.githubusercontent.com/4188758/136195623-d856d5b0-72e6-4b31-8d21-386506ed3c18.png)

**Reply by MichailiK** [reply]
I've submitted [JMusicBot-0.3.6.jar](https://github.com/jagrosh/MusicBot/releases/download/0.3.6/JMusicBot-0.3.6.jar) (SHA-256 `1dcdddb0e0030be8d5652718d7d3980e41e29a7b4c8ddc3f0680771144bf7749`) as a false-positive to Bitdefender, hoping to get JMusicBot unflagged.

**Reply by jkiller295** [reply]
> I've submitted [JMusicBot-0.3.6.jar](https://github.com/jagrosh/MusicBot/releases/download/0.3.6/JMusicBot-0.3.6.jar) (SHA-256 `1dcdddb0e0030be8d5652718d7d3980e41e29a7b4c8ddc3f0680771144bf7749`) as a false-positive to Bitdefender, hoping to get JMusicBot unflagged.

Please inform when Bitdefender approves your submission. I'd like to use your bot.

**Comment by samhwang** [comment]
Also scanned this repo via Snyk, I found this error in the logback core module. But I think it's not related to the reported Trojan above?

https://snyk.io/vuln/SNYK-JAVA-CHQOSLOGBACK-1726923

![Screen Shot 2021-10-06 at 9 20 36 am](https://user-images.githubusercontent.com/4188758/136110813-5ba8e551-5420-41b1-8980-31a1f3f8e5b6.png)

**Reply by MichailiK** [reply]
We use logback for logs in JMusicBot. One of logback's features is to send logs over the internet. Because all logging happens locally, this vulnerability does not affect JMusicBot.

**Reply by Sanduhr32** [reply]
I believe logback 1.3.2 has this fixed

**Comment by MichailiK** [comment, marked as answer]
I've contacted Bitdefender and it looks like they (among almost all other AVs that have flagged JMusicBot) have updated their signatures to no longer falsely detect JMusicBot as malicious software.

**Reply by jkiller295** [reply]
I can confirm Bitdefender no longer flags the .jar file as a Trojan. However, MS Edge still gives you warning "This file might harm blah blah" on download

**Reply by jagrosh** [reply]
I would assume that message would be standard for downloading *any* executable jar file

**Reply by YJSoft** [reply]
@jkiller295 I think that message is generated based on MS's `SmartScreen Reputation`. since reputation is based on code sign certificate's id(which is very expensive), that will unlikely to solved itself. Anyone willing download jar file can bypass that message by pressing continue.

**Reply by MichailiK** [reply]
> @jkiller295 I think that message is generated based on MS's `SmartScreen Reputation`.

This is false. Looking at Chromium [source code](https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/resources/download_file_types.asciipb;l=743), **all `.jar` files** are flagged as Dangerous and trigger a "This file might harm your computer" warning.

See:
- <https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/resources/safe_browsing/README.md>
- <https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/resources/download_file_types.asciipb;l=743>, lines 742-755
